File: Machine Learning A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression_myfile.R
setwd("C:\\Users\\toshiba\\Desktop\\Machine Learning A-Z\\Part 2 - Regression\\Section 4 - Simple Linear Regression")

# Simple Linear Regression
dataset = read.csv('Kidem_ve_Maas_VeriSeti.csv')
# dataset = dataset[,2:3]

# Splitting the dataset into the training and test
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Maas, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3] = scale(test_set[,2:3])

# Fitting Simple Linear Regression to the Training Set
regressor = lm(formula = Maas ~ Kidem, data = training_set)
summary(regressor)

# Predicting the Test set results
y_pred = predict(regressor, newdata = test_set)
View(y_pred)

# Visualising the Training set results
library(ggplot2)
ggplot()+
  geom_point(aes(x = training_set$Kidem, y = training_set$Maas),
             color='red')+
  geom_line(aes(x = training_set$Kidem, y = predict(regressor, newdata = training_set)),
            color = 'blue')+
  ggtitle('Kıdeme Göre Maaş Tahmini Regresyon Model Grafiği')+
  xlab('Kıdem(Yıl)')+
  ylab('Maaş(Yıllık - TL)')

# Visualising the Test set results
ggplot()+
  geom_point(aes(x = test_set$Kidem, y = test_set$Maas),
             color='red')+
  geom_line(aes(x = training_set$Kidem, y = predict(regressor, newdata = training_set)),
            color = 'blue')+
  ggtitle('Kıdeme Göre Maaş Tahmini Regresyon Model Grafiği (Test Seti)')+
  xlab('Kıdem(Yıl)')+
  ylab('Maaş(Yıllık - TL)')

File: Machine Learning A-Z/Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression_myfile.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jun 13 00:16:26 2017

@author: toshiba
"""
# Simpler Linear Regression
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import os
os.chdir('C:\\Users\\toshiba\\Desktop\\Machine Learning A-Z\\Part 2 - Regression\\Section 4 - Simple Linear Regression')
# Importing the dataset
dataset = pd.read_csv('Kidem_ve_Maas_VeriSeti.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""

# Fitting Simple Linear Regression to training model
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

# Predicting the Test set results
y_pred = regressor.predict(X_test)

# Visualising the Training set results
plt.scatter(X_train, y_train, color = 'red')
modelin_tahmin_ettigi_y = regressor.predict(X_train)
plt.plot(X_train, modelin_tahmin_ettigi_y, color = 'blue')
plt.title('Kıdeme Göre Maaş Tahmini Regresyon Modeli')
plt.xlabel('Kıdem')
plt.ylabel('Maaş')
plt.show()

# Visualising the Test set results
plt.scatter(X_test, y_test, color = 'red')
modelin_tahmin_ettigi_y = regressor.predict(X_train)
plt.plot(X_train, modelin_tahmin_ettigi_y, color = 'blue')
plt.title('Kıdeme Göre Maaş Tahmini Regresyon Modeli')
plt.xlabel('Kıdem')
plt.ylabel('Maaş')
plt.show()